music_assistant/providers/subsonic_scrobble/__init__.py

```diff
@@ -0,0 +1,167 @@
+        """
+        Handle unload/close of the provider.
+
+        Called when provider is deregistered (e.g. MA exiting or config reloading).
+        """
+        for unload_cb in self._on_unload:
+            unload_cb()
+
+
+class SubsonicScrobbleEventHandler(ScrobblerHelper):
+    """Handles the scrobbling event handling."""
+
+    def __init__(self, mass: MusicAssistant, logger: logging.Logger) -> None:
+        """Initialize."""
+        super().__init__(logger)
+        self._mass = mass
+
+    def _is_scrobblable_media_type(self, media_type: MediaType) -> bool:
+        """Return true if the given OpenSubsonic media type can be scrobbled, false otherwise."""
+        return media_type in (
+            MediaType.TRACK,
+            MediaType.AUDIOBOOK,
+            MediaType.PODCAST_EPISODE,
+        )
+
+    async def _get_subsonic_provider_and_item_id(
+        self, media_type: MediaType, provider_instance_id_or_domain: str, item_id: str
+    ) -> tuple[None | OpenSonicProvider, str]:
+        """Return a OpenSonicProvider or None if no subsonic provider, and the Subsonic item_id.
+
+        Returns:
+            Tuple[OpenSonicProvider | None, str]: The provider or None, and the Subsonic item_id.
+        """
+        if provider_instance_id_or_domain == "library":
+            # unwrap library item to check if we have a subsonic mapping...
+            library_item = await self._mass.music.get_library_item_by_prov_id(
+                media_type, item_id, provider_instance_id_or_domain
+            )
+            if library_item is None:
+                return None, item_id
+            assert isinstance(library_item, (Track, Audiobook, PodcastEpisode))
+            for mapping in library_item.provider_mappings:
+                if mapping.provider_domain.startswith("opensubsonic"):
+                    # found a subsonic mapping, proceed...
+                    prov = self._mass.get_provider(mapping.provider_instance)
+                    assert isinstance(prov, OpenSonicProvider)
+                    return prov, mapping.item_id
+            # no subsonic mapping has been found in library item, ignore...
+            return None, item_id
+        elif provider_instance_id_or_domain.startswith("opensubsonic"):
+            # found a subsonic mapping, proceed...
+            prov = self._mass.get_provider(provider_instance_id_or_domain)
+            assert isinstance(prov, OpenSonicProvider)
+            return prov, item_id
+        # not an item from subsonic provider, ignore...
+        return None, item_id
+
+    async def _update_now_playing(self, report: MediaItemPlaybackProgressReport) -> None:
+        def handler(prov: OpenSonicProvider, item_id: str, uri: str) -> None:
+            try:
+                self.logger.info("scrobble play now event")
+                prov._conn.scrobble(item_id, submission=False)
+                self.logger.debug(f"track {uri} marked as 'now playing'")
```

Comment:
```suggestion
                self.logger.debug("track %s marked as 'now playing'", uri)
```

prevent using f-strings in log statements

Reply:
Changed to string formatting.